src/components/app/details/AboutTagEditModal.tsx

```diff
@@ -0,0 +1,128 @@
+        }
+    }
+    const outsideClickHandler = (evt): void => {
+        if (editLabelRef.current && !editLabelRef.current.contains(evt.target) && typeof onClose === 'function') {
+            onClose(evt)
+        }
+    }
+
+    useEffect(() => {
+        document.addEventListener('keydown', escKeyPressHandler)
+        return (): void => {
+            document.removeEventListener('keydown', escKeyPressHandler)
+        }
+    }, [escKeyPressHandler])
+
+    useEffect(() => {
+        document.addEventListener('click', outsideClickHandler)
+        return (): void => {
+            document.removeEventListener('click', outsideClickHandler)
+        }
+    }, [outsideClickHandler])
+
+    const handleSaveAction = async (e): Promise<void> => {
+        e.preventDefault()
+        const _labelTags = []
+        let invalidLabels = false
+        for (let index = 0; index < labelTags.length; index++) {
+            const element = labelTags[index]
+            if (element.isInvalidKey || element.isInvalidValue) {
+                invalidLabels = true
+                break
+            } else if (element.key) {
+                _labelTags.push({ key: element.key, value: element.value, propagate: element.propagate })
+            }
+        }
+        if (invalidLabels) {
+            toast.error('Some required fields are missing or invalid')
+            return
+        }
+        setSubmitting(true)
+
+        const payload = {
+            id: parseInt(appId),
+            labels: _labelTags,
+        }
+
+        try {
+            await createAppLabels(payload)
+            toast.success('Successfully saved')
+            // Fetch the latest project & labels details
+            await getAppMetaInfoRes()
+        } catch (err) {
+            showError(err)
+        } finally {
+            onClose(e)
+            setSubmitting(false)
+        }
+    }
+
+    const renderAboutModalInfo = (): JSX.Element => {
+        return (
+            <>
+                <div className="cn-7 p-20 dc__overflow-scroll" style={{ height: 'calc(100vh - 122px)' }}>
+                    <TagLabelSelect labelTags={labelTags} setLabelTags={setLabelTags} />
+                </div>
+                <div className="form__buttons dc__border-top pt-16 pb-16 pl-20 pr-20">
+                    <button
+                        className="cta cancel flex h-36 mr-12"
+                        type="button"
+                        disabled={submitting}
+                        onClick={onClose}
+                        tabIndex={6}
+                    >
+                        Cancel
+                    </button>
+                    <button
+                        className="cta flex h-36"
+                        type="submit"
+                        disabled={submitting}
+                        onClick={handleSaveAction}
+                        tabIndex={5}
+                    >
+                        {submitting ? <Progressing /> : 'Save'}
```

Comment:
**Just a suggestion** : Logically Here we are updating the value so should we keep `Save` as an `Update` and state `submitting` as an `updating`

Reply:
It is as per the UI